tests/store/test_runs_artifact_repo.py

```diff
@@ -0,0 +1,47 @@
+import mock
+import pytest
+
+from mlflow.exceptions import MlflowException
+from mlflow.store.runs_artifact_repo import RunsArtifactRepository
+
+
+@pytest.mark.parametrize("uri, expected_run_id, expected_artifact_path", [
+    ('runs:/1234abcdf1394asdfwer33/path/to/model', '1234abcdf1394asdfwer33', 'path/to/model'),
+    ('runs:/1234abcdf1394asdfwer33/path/to/model/', '1234abcdf1394asdfwer33', 'path/to/model/'),
+    ('runs:/1234abcdf1394asdfwer33', '1234abcdf1394asdfwer33', None),
+    ('runs:/1234abcdf1394asdfwer33/', '1234abcdf1394asdfwer33', None),
```

Comment:
BTW, thanks for these tests - the parsing logic in RunsArtifactRepository was kinda fancy so it's good to have these cases to confirm things work. I was kinda wondering if it'd be better to use a regex to parse the URIs instead of `urllib` but the current approach seems better (otherwise we run into all sorts of complications around making sure the regex matches the URI spec)

Reply:
lol yeah i couldn't get the logic right without tests xD